moped-api/users/users.py

```diff
@@ -0,0 +1,163 @@
+import os, boto3
+from botocore.exceptions import ClientError
+from flask import Blueprint, jsonify, abort
+from flask_cognito import cognito_auth_required, current_cognito_jwt, request
+from config import api_config
+
+# Import our custom code
+from claims import normalize_claims, get_claims
+from users.helpers import (
+    load_claims,
+    put_claims,
+    format_claims,
+    is_valid_user,
+    has_user_role,
+)
+
+users_blueprint = Blueprint("users_blueprint", __name__)
+
+MOPED_API_CURRENT_ENVIRONMENT = os.getenv("MOPED_API_CURRENT_ENVIRONMENT", "STAGING")
+USER_POOL = api_config[MOPED_API_CURRENT_ENVIRONMENT]["COGNITO_USERPOOL_ID"]
+
+
+@users_blueprint.route("/", methods=["GET"])
+@cognito_auth_required
+def user_list_users() -> str:
+    """
+    Returns users in user pool
+    :return str:
+    """
+    if is_valid_user(current_cognito_jwt):
+        cognito_client = boto3.client("cognito-idp")
+
+        user_response = cognito_client.list_users(UserPoolId=USER_POOL)
+        user_list = user_response["Users"]
+        return jsonify(user_list)
+    else:
+        abort(403)
+
+
+@users_blueprint.route("/<id>", methods=["GET"])
+@cognito_auth_required
+def user_get_user(id: str) -> str:
+    """
+    Returns user details
+    :return str:
+    """
+    if is_valid_user(current_cognito_jwt):
+        cognito_client = boto3.client("cognito-idp")
+
+        user_dict = {}
+
+        user_info = cognito_client.admin_get_user(UserPoolId=USER_POOL, Username=id)
+        user_roles = load_claims(id)
+
+        user_dict.update(user_info)
+        user_dict.update(user_roles)
+
+        return jsonify(user_dict)
+    else:
+        abort(403)
+
+
+@users_blueprint.route("/", methods=["POST"])
+@cognito_auth_required
+@normalize_claims
+def user_create_user(claims: list) -> str:
+    """
+    Returns created user details
+    :return str:
+    """
+    if is_valid_user(current_cognito_jwt) and has_user_role("user", claims):
+        cognito_client = boto3.client("cognito-idp")
```

Comment:
This line repeats itself a lot in this file, would it not be better to declare it once globally and let function access the cognito_client?